view/GameView/PlayingFieldScreen.py

```diff
@@ -123,11 +125,36 @@ def update_playingfield(self) -> None:
                 if field.getGadget().has_value():
                     self.map.map[WorldPoint(x, y, z=1)] = Gadget(WorldPoint(x, y, z=1), self.asset_storage)
 
+                # check if field is foggy
+                if field.isFoggy():
+                    self.map.map[WorldPoint(x, y, z=1)] = Fog(WorldPoint(x, y, z=1), self.asset_storage)
+
         # add characters
         for char in state.getCharacters():
             if not char.getCoordinates().has_value():
                 continue
             point = char.getCoordinates().value()
-
             self.map.map[WorldPoint(point.x, point.y, z=1)] = Character(WorldPoint(point.x, point.y, z=1),
                                                                         self.asset_storage)
+        # check if janitor is on playing field
+        if state.getJanitorCoordinates().has_value():
+            pos_cpp = state.getJanitorCoordinates().value()
+            self.map.map[WorldPoint(pos_cpp.x, pos_cpp.y, z=1)] = Janitor(WorldPoint(pos_cpp.x, pos_cpp.y, z=1),
+                                                                          self.asset_storage)
+
+        # check if cat is on playing field
+        if state.getCatCoordinates().has_value():
+            pos_cpp = state.getCatCoordinates().value()
+            self.map.map[WorldPoint(pos_cpp.x, pos_cpp.y, z=1)] = Cat(WorldPoint(pos_cpp.x, pos_cpp.y, z=1),
+                                                                      self.asset_storage)
+
+        # mark active char red.
+        active_char = self.controller.lib_client_handler.lib_client.getActiveCharacter()
+        active_char_coords = self.controller.lib_client_handler.lib_client.getState().getCharacters().findByUUID(
+            active_char).getCoordinates().value()
+        wp_coords = WorldPoint(active_char_coords.x, active_char_coords.y, 0)
+        self.map.map[wp_coords].active_char()
+        self.map.map[WorldPoint(active_char_coords.x, active_char_coords.y, 2)] = Floor(
+            WorldPoint(active_char_coords.x, active_char_coords.y, 2), self.asset_storage)
+        self.map.map[WorldPoint(active_char_coords.x, active_char_coords.y, 2)].active_char()
```

Comment:
Das hier könnte man noch mit in die Methode `active_char()` machen, weil eigentlich gehört dass ja da dazu und kleine optische Verbesserungsidee (optional, muss auch nicht in diesem pr gemacht werden): statt einem Floor könnte man einen Pfeil über dem Charakter (braucht man halt noch ein asset) schweben lassen.
Evtl. könnten die Aktualisierungsprobleme von dem Pfeil/Floor auch gelöst werden, indem es dieses "Objekt" nur einmal gibt und in der  `active_char()` Methode an die richtige Position geschoben wird?

Reply:
Den Pfeil wollte ich machen, hab aber keinen gefunden